app/ami.py

```diff
@@ -0,0 +1,158 @@
+                amis.append(ami)
+        return amis
+
+    # Get the info about a single AMI
+    @staticmethod
+    def build_model(region_name: str, resource_dict: dict):
+        tags = resource.make_tags_dict(resource_dict.get('Tags', []))
+
+        state = resource_dict['State']
+        ec2_type = 'ami'
+
+        resource_id_tag = 'ImageId'
+        resource_type_tag = 'ImageType'
+        name = resource_dict[resource_id_tag]
+        ami = Resource.build_generic_model(tags, resource_dict, region_name, resource_id_tag, resource_type_tag)
+        ami_type = resource_dict['RootDeviceType']  # either instance-store or ebs
+        block_device_mappings = resource_dict['BlockDeviceMappings']  # the collection of ebs snapshots forming the ami
+
+        monthly_price = resource.estimate_monthly_ami_price(ami_type, block_device_mappings, name)
+
+        return Ami(region_name=region_name,
+                      resource_id=ami.resource_id,
+                      state=state,
+                      reason=ami.reason,
+                      resource_type=ami.resource_type,
+                      ec2_type=ec2_type,
+                      eks_nodegroup_name=ami.eks_nodegroup_name,
+                      name=ami.name,
+                      operating_system=ami.operating_system,
+                      monthly_price=monthly_price,
+                      stop_after=ami.stop_after,
+                      terminate_after=ami.terminate_after,
+                      nagbot_state=ami.nagbot_state,
+                      contact=ami.contact,
+                      stop_after_tag_name=ami.stop_after_tag_name,
+                      terminate_after_tag_name=ami.terminate_after_tag_name,
+                      nagbot_state_tag_name=ami.nagbot_state_tag_name,
+                      iops=ami.iops,
+                      throughput=ami.throughput)
+
+    # Delete/terminate an AMI
+    def terminate_resource(self, dryrun: bool) -> bool:
+        print(f'Deleting AMI: {str(self.resource_id)}...')
+        ec2 = boto3.resource('ec2', region_name=self.region_name)
+        image = ec2.Image(self.resource_id)
+        try:
+            if not dryrun:
+                response = image.deregister()  # response should return None
+                print(f'Response from image.deregister() (should be None): {str(response)}')
+            return True
+        except Exception as e:
+            print(f'Failure when calling image.deregister(): {str(e)}')
+            return False
+
+    def is_stoppable_without_warning(self):
+        return self.generic_is_stoppable_without_warning(self)
+
+    # Check if an ami is stoppable (should always be false)
+    def is_stoppable(self, today_date, is_weekend=TODAY_IS_WEEKEND):
+        return self.generic_is_stoppable(self, today_date, is_weekend)
+
+    # Check if an ami is deletable/terminatable
+    def is_terminatable(self, today_date):
+        state = 'available'
```

Comment:
Yes, that information is included in the "Description" of the dictionary returned by the aws call - I just added two additional attributes in the Snapshot object, `is_ami_snapshot` and `is_aws_backup` snapshot. If the snapshot is either it will not get deleted, and I adjusted the pytests to test for that scenario.